src/new/wallet.js

```diff
@@ -1339,6 +1341,46 @@ class HathorWallet extends EventEmitter {
     this.conn.stop();
   }
 
+  /**
+   * Returns an address' privateKey given an index and the encryption password
+   *
+   * @param {string} pinCode - The PIN used to encrypt data in accessData
+   * @param {number} addressIndex - The address' index to fetch
+   *
+   * @memberof HathorWalletServiceWallet
+   * @inner
+   */
+  async getAddressPrivKey(pinCode, addressIndex) {
+    const mainXPrivKey = await this.storage.getMainXPrivKey(pinCode);
+    const addressXPriv = new bitcore.HDPrivateKey(mainXPrivKey).derive(addressIndex);
+
+    return addressXPriv.privateKey;
```

Comment:
Why not return the `HDPrivateKey`? this way we will have access to both private and public keys if needed